hie-compat/src-ghc86/Compat/HieUtils.hs

```diff
@@ -126,14 +126,13 @@ recoverFullType i m = go i
     go i = Roll $ fmap go (m A.! i)
 
 getTypeIndex :: Type -> State HieTypeState TypeIndex
-getTypeIndex t
-  | otherwise = do
-      tm <- gets tyMap
-      case lookupTypeMap tm t of
-        Just i -> return i
-        Nothing -> do
-          ht <- go t
-          extendHTS t ht
+getTypeIndex t = do
+    tm <- gets tyMap
+    case lookupTypeMap tm t of
+      Just i -> return i
+      Nothing -> do
+        ht <- go t
+        extendHTS t ht
```

Comment:
I think the point of hie-compat library is using the original code as-is. The original `ghc` library code itself contains this redundant `otherwise`, I think it's better to keep it as-is.

Reply:
Alright, I reverted all changes within hie-compat module.